version: "3.8"
services:
  mysqldb:
    image: mysql:8.0
    restart: unless-stopped 
    env_file: ./.env
    environment:
      MYSQL_ROOT_PASSWORD: ${MYSQL_DB_PASSWORD} 
      MYSQL_DATABASE: fitdb
    ports:
      - "3307:3306"    
    healthcheck:
       test: ["CMD", "curl", "-f", "http://localhost:3306"]
       interval: 30s
       timeout: 10s
       retries: 5
    volumes:
      - db:/var/lib/mysql
      - ./db/init.sql:/docker-entrypoint-initdb.d/data_dump.sql
    networks:
        - backend

  fit-api:
    depends_on:
      - mysqldb 
    build: ./fit-api
    restart: on-failure 
    env_file: ./.env
    ports:
      - ${NODE_LOCAL_PORT}:${NODE_DOCKER_PORT}
    environment:
      DB_HOST: mysqldb
      DB_PASSWORD: ${MYSQL_DB_PASSWORD}
      DB_NAME: ${MYSQL_DB_NAME}
      DB_PORT: ${MYSQL_PORT}
      DB_USER: ${MYSQL_DB_USER}
      CLIENT_ORIGIN: ${CLIENT_ORIGIN}
      SECRET: ${SECRET}
    networks:
        - backend
        - frontend
    volumes:
      - ./fit-api:/fit-api
      - /fit-api/node_modules
  frontend:
    depends_on:
      - fit-api
    build: 
      context: ./fit-app
      args: 
        - REACT_APP_API_BASE_URL:${CLIENT_API_BASE_URL}
        - NEXT_APP_API_BASE_URL:${CLIENT_API_BASE_URL}
    container_name: fitapp-react_c
    ports:
      - '3000:3000'
    volumes:
      - ./fit-app:/fit-app
      - /fit-app/node_modules
volumes:
  db:
networks:
  backend:
  frontend:
